packages/adapter-neon/src/neon.test.ts

```diff
@@ -26,3 +26,44 @@ describe('validation', () => {
     }).not.toThrow()
   })
 })
+
+describe('neon version support', () => {
+  it('supports @neondatabase/serverless < 1.0.0', async () => {
+    // Before v0 client is directly callable to execute the SQL
+    const v0NeonClientMock = jest
+      .fn()
+      .mockResolvedValue({ fields: [], command: '', rowCount: 0, rows: [], rowAsArray: false })
```

Comment:
Why would we test a mock, rather than testing how the actual `@neondatabase/serverless` client works?

We can store multiple versions of it in `package.json`'s `devDependencies`, as long as we mark them via aliases:

```json
{
  "devDependencies": {
    "serverless-v1": "npm:@neondatabase/serverless@^0.10.0",
    "serverless-v2": "npm:@neondatabase/serverless@^1.0.0"
  }
}
```

Reply:
Oh - I wasn't aware you could even do that. Will adjust it that way :+1:

Reply:
I actually can't improve the test much in terms of not mocking things. If I want to fully avoid mocking we'd have to connect to real neon database here (or a containerized setup using the neon-proxy).

I think that's not worth the overhead and effort here. I rather instead update our neon ecosystem tests to use neon v1 once it is released.